lms/djangoapps/courseware/tests/test_entrance_exam.py

```diff
@@ -206,6 +211,100 @@ def setUp(self):
             ]
         )
 
+    @mock.patch('xmodule.x_module.XModuleMixin.has_dynamic_children', mock.Mock(return_value='True'))
+    def test_view_redirect_if_entrance_exam_required(self):
```

Comment:
@cahrens -- the gating logic is designed to unlock the rest of the course after you have acheived the minimum score required -- so if the minimum score is 50%, and there are two problems in the exam, and you answer one correctly, the rest of the course is now available to you. @marcotuts -- that is how it's supposed to work, correct?